### PR TITLE
[Core] Support create and delete branch

### PR DESCRIPTION
**Pull request by TaoZex**
### Purpose

Introduce branch for paimon
Linked issue: https://github.com/apache/incubator-paimon/issues/2148
deisgn：https://cwiki.apache.org/confluence/display/PAIMON/PIP-9%3A+Support+Branch
task：https://github.com/apache/incubator-paimon/issues/1795

### CODE REVIEW

**Comment by TaoZex** [comment]
https://github.com/apache/incubator-paimon/pull/2608
After communicate with @FangYongs ,only introduce the BranchManager to support create and delete branch in this pr. For merge and replace branch, will be committed in the next pr.

File: paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java [hunk 1]
```diff
@@ -471,6 +480,17 @@ public Path toSchemaPath(long id) {
         return new Path(tableRoot + "/schema/" + SCHEMA_PREFIX + id);
     }
 
+    public Path branchSchemaPath(String branchName, long schemaId) {
+        return new Path(
+                tableRoot
+                        + "/branch/"
+                        + BRANCH_PREFIX
+                        + branchName
```
Comment (FangYongs) [inline paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java:488]: It's better to use `BranchManager#getBranchPath` to build path for branch here?

File: paimon-core/src/main/java/org/apache/paimon/utils/SnapshotManager.java [hunk 2]
```diff
@@ -80,6 +81,17 @@ public Path snapshotPath(long snapshotId) {
         return new Path(tablePath + "/snapshot/" + SNAPSHOT_PREFIX + snapshotId);
     }
 
+    public Path branchSnapshotPath(String branchName, long snapshotId) {
+        return new Path(
+                tablePath
+                        + "/branch/"
+                        + BRANCH_PREFIX
+                        + branchName
```
Comment (FangYongs) [inline paimon-core/src/main/java/org/apache/paimon/utils/SnapshotManager.java:89]: As above

File: paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java [hunk 3]
```diff
@@ -66,6 +67,18 @@ public Path tagPath(String tagName) {
         return new Path(tablePath + "/tag/" + TAG_PREFIX + tagName);
     }
 
+    /** Return the path of a tag in branch. */
+    public Path branchTagPath(String branchName, String tagName) {
+        return new Path(
+                tablePath
+                        + "/branch/"
+                        + BRANCH_PREFIX
+                        + branchName
```
Comment (FangYongs) [inline paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java:76]: As above

**Comment by FangYongs** [comment]
Thanks @TaoZex , just minor comment

**Comment by TaoZex** [comment]
@FangYongs Thanks for your review, I have modified the code.

**Comment by TaoZex** [comment]
Timeout in ci. Please rerun it.

**Pull request merged**